common/src/main/java/com/wynntils/features/DiscordRichPresenceFeature.java

```diff
@@ -0,0 +1,154 @@
+/*
+ * Copyright © Wynntils 2023.
+ * This file is released under AGPLv3. See LICENSE for full license details.
+ */
+package com.wynntils.features;
+
+import com.wynntils.core.components.Managers;
+import com.wynntils.core.components.Models;
+import com.wynntils.core.config.Config;
+import com.wynntils.core.config.ConfigHolder;
+import com.wynntils.core.config.RegisterConfig;
+import com.wynntils.core.features.Feature;
+import com.wynntils.core.text.PartStyle;
+import com.wynntils.core.text.StyledText;
+import com.wynntils.mc.event.ConnectionEvent;
+import com.wynntils.mc.event.SetXpEvent;
+import com.wynntils.models.character.event.CharacterUpdateEvent;
+import com.wynntils.models.character.type.ClassType;
+import com.wynntils.models.territories.profile.TerritoryProfile;
+import com.wynntils.models.worlds.event.WorldStateEvent;
+import com.wynntils.utils.mc.McUtils;
+import java.util.Locale;
+import net.minecraft.core.Position;
+import net.minecraftforge.eventbus.api.SubscribeEvent;
+
+public class DiscordRichPresenceFeature extends Feature {
+    @RegisterConfig
+    public final Config<Boolean> displayLocation = new Config<>(true);
+
+    @RegisterConfig
+    public final Config<Boolean> displayCharacterInfo = new Config<>(true);
+
+    @RegisterConfig
+    public final Config<Boolean> displayWorld = new Config<>(true);
+
+    private boolean stopTerritoryCheck = false;
+    private TerritoryProfile lastTerritoryProfile = null;
+    private ClassType classType = null;
+    private int level = 0;
+
+    private void setCharacterDetails() {
+        if (classType == null) return;
+        String name = StyledText.fromComponent(McUtils.player().getName()).getString(PartStyle.StyleType.NONE);
+        Managers.Discord.setImageText(name + " - Level " + level + " " + classType.getName());
+        Managers.Discord.setImage(classType.getActualName(false).toLowerCase(Locale.ROOT));
+    }
+
+    @SubscribeEvent
+    public void onCharacterUpdate(CharacterUpdateEvent event) {
+        if (!Models.WorldState.onWorld()) return;
+
+        // classType needs to be set even when config is disabled so if the config is enabled later, it will not require
+        // a relog
```

Comment:
Why is that? You are calling setCharacterDetails() from onConfigChange.

Reply:
setCharacterDetails() doesn't set the feature's classType, it just tells the rich presence to display what is already stored. This is a misleading name now that you brought it up, I will change it.

Reply:
I was more wondering why you need a classType field at all, and not just read it from the model whenever you need to change the class info?

Reply:
Originally it was so that when the display function was called it wouldn't have to ask the model for the classType as much, but I suppose both are reading some field and that's it, so it would be the same for performance